app/api/groups/get_group.go

```diff
@@ -0,0 +1,38 @@
+package groups
+
+import (
+	"errors"
+	"fmt"
+	"github.com/go-chi/render"
+	"net/http"
+
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+func (srv *Service) getGroup(w http.ResponseWriter, r *http.Request) service.APIError {
+	groupID, err := service.ResolveURLQueryPathInt64Field(r, "group_id")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	user := srv.GetUser(r)
+
+	query := srv.Store.GroupAncestors().OwnedByUserID(user.UserID).
+		Joins("JOIN groups ON idGroupChild=groups.ID").
+		Where("idGroupChild = ?", groupID).Select(
```

Comment:
for the pure logic point of view (so readability as well), I would have expected `groups` as root table. So have `srv.Store.Groups().OwnedBy(user).Where(groups.ID = ?, groupID)`  with the `OwnedBy(...)` doing the "matching," e.g. inner-joining with group ancestors and only caring for the "left" part

Reply:
fixed